radixdlt-java/src/main/java/com/radixdlt/client/core/network/actions/SubmitAtomResultAction.java

```diff
@@ -0,0 +1,92 @@
+package com.radixdlt.client.core.network.actions;
+
+import com.google.gson.JsonElement;
+import com.radixdlt.client.core.atoms.Atom;
+import com.radixdlt.client.core.network.RadixNode;
+import com.radixdlt.client.core.network.jsonrpc.RadixJsonRpcClient.NodeAtomSubmissionState;
+import com.radixdlt.client.core.network.jsonrpc.RadixJsonRpcClient.NodeAtomSubmissionUpdate;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * A dispatchable event action which signifies the end result of an atom submission flow
+ */
+public class SubmitAtomResultAction implements SubmitAtomAction {
+	public enum SubmitAtomResultActionType {
+		FAILED(NodeAtomSubmissionState.FAILED),
```

Comment:
Yes, I like this proposal (and I actually do like a good ole switch statement :) though I don't think there are compiler warnings if there are missing case statements since the compiler doesn't know if there may be additional enums during runtime: https://stackoverflow.com/questions/39225553/complete-switch-on-enum-in-java-produces-missing-return-statement-error